### DISCUSSION TITLE
Microsoft Visual C++ 2015-2022 (x64) version downloading x86 version

### DISCUSSION THREAD

**Discussion by ja-law** [Romanitho/Winget-AutoUpdate#568] [General]
Hi, anyone else seen winget pulling down the x86 version of vcredist for the x64 version. Extract from the WAU log:

10:57:42 - Checking application updates on Winget Repository...
-> Available update : Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.38.. Current version : 14.38.33130.0. Available version : 14.38.33135.0.
10:57:46 - Updating Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.38. from 14.38.33130.0 to 14.38.33135.0...
10:57:51 - ##########   WINGET UPGRADE PROCESS STARTS FOR APPLICATION ID 'Microsoft.VCRedist.2015+.x86'   ##########
10:57:51 - -> Running: Winget upgrade --id Microsoft.VCRedist.2015+.x86 -e --accept-package-agreements --accept-source-agreements -s winget -h

Thanks

Jason

**Comment by AndrewDemski-ad-gmail-com** [comment]
I just checked my logs for that particular upgrade..

I haven't observed such behavior.

> ################################################################
> `#     2024-02-06 - CHECK FOR APP UPDATES (System context - Connected user)
> ################################################################
> 16:04:39 - Notification Level: Full. Notification Language: Polski
> 16:04:39 - Checking internet connection...
> 16:04:39 - Connected!
> 16:04:39 - Winget Version: v1.6.3482
> 16:04:39 - WAU current version: 1.19.1
> 16:04:39 - WAU AutoUpdate is Enabled.
> 16:04:39 - WAU is up to date.
> 16:04:39 - WAU uses Black List config
> 16:04:39 - Checking application updates on Winget Repository...
> -> Available update : Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.38.. Current version : 14.38.33130.0. Available version : 14.38.33135.0.
> 16:04:46 - Updating Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.38. from 14.38.33130.0 to 14.38.33135.0...
> 16:04:50 - ##########   WINGET UPGRADE PROCESS STARTS FOR APPLICATION ID 'Microsoft.VCRedist.2015+.x64'   ##########
> 16:04:50 - -> Running: Winget upgrade --id Microsoft.VCRedist.2015+.x64 -e --accept-package-agreements --accept-source-agreements -s winget -h
> Found Microsoft Visual C++ 2015-2022 Redistributable (x64) [Microsoft.VCRedist.2015+.x64] Version 14.38.33135.0
> This application is licensed to you by its owner.
> Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
> Downloading https://download.visualstudio.microsoft.com/download/pr/6ba404bb-6312-403e-83be-04b062914c98/1AD7988C17663CC742B01BEF1A6DF2ED1741173009579AD50A94434E54F56073/VC_redist.x64.exe
> Successfully verified installer hash
> Starting package install...
> Successfully installed
> 16:05:27 - ##########   WINGET UPGRADE PROCESS FINISHED FOR APPLICATION ID 'Microsoft.VCRedist.2015+.x64'   ##########
> 16:05:27 - Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.38. updated to 14.38.33135.0 !
> 16:05:31 - 1 apps updated ! No more update.

**Reply by ja-law** [reply]
Ok , not sure what's going on then

**Reply by Boesh89** [reply]
Hello

I'm experiencing exactly the same:
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/102753813/fe7f09a3-6026-4d0c-aea4-8c2ea0988f5d)

As you can see, the script is trying to update the Microsoft Visual C++ 2015-2022 Redistributable (x64) version but using the x86 Winget ID.
